drivers/at86rf2xx/at86rf2xx_rf_ops.c

```diff
@@ -0,0 +1,971 @@
+    }
+
+    if (!buf) {
+        at86rf2xx_fb_stop(dev);
+        mutex_unlock(&dev->lock);
+        return 0;
+    }
+
+    /* copy payload */
+    at86rf2xx_fb_read(dev, (uint8_t *)buf, pkt_len);
+
+    /* Ignore FCS but advance fb read - we must give a temporary buffer here,
+     * as we are not allowed to issue SPI transfers without any buffer */
+    uint8_t tmp[2];
+    at86rf2xx_fb_read(dev, tmp, 2);
+    (void)tmp;
+
+    /* AT86RF212B RSSI_BASE_VAL + 1.03 * ED, base varies for diff. modulation and datarates
+     * AT86RF232  RSSI_BASE_VAL + ED, base -91dBm
+     * AT86RF233  RSSI_BASE_VAL + ED, base -94dBm
+     * AT86RF231  RSSI_BASE_VAL + ED, base -91dBm
+     * AT86RFA1   RSSI_BASE_VAL + ED, base -90dBm
+     * AT86RFR2   RSSI_BASE_VAL + ED, base -90dBm
+     *
+     * AT86RF231 MAN. p.92, 8.4.3 Data Interpretation
+     * AT86RF232 MAN. p.91, 8.4.3 Data Interpretation
+     * AT86RF233 MAN. p.102, 8.5.3 Data Interpretation
+     *
+     * for performance reasons we ignore the 1.03 scale factor on the 212B,
+     * which causes a slight error in the values, but the accuracy of the ED
+     * value is specified as +/- 5 dB, so it should not matter very much in real
+     * life.
+     */
+    if (info != NULL) {
+        uint8_t ed = 0;
+        at86rf2xx_fb_read(dev, &(info->lqi), 1);
+
+#if AT86RF2XX_HAVE_ED_REGISTER
+        /* AT86RF231 does not provide ED at the end of the frame buffer, read
+         * from separate register instead */
+        at86rf2xx_fb_stop(dev);
+        ed = at86rf2xx_reg_read(dev, AT86RF2XX_REG__PHY_ED_LEVEL);
+#else
+        at86rf2xx_fb_read(dev, &ed, 1);
+        at86rf2xx_fb_stop(dev);
+#endif
+        info->rssi = RSSI_BASE_VAL + ed;
+        DEBUG("[at86rf2xx] LQI:%d high is good, RSSI:%d high is either good or"
+              "too much interference.\n", info->lqi, info->rssi);
+    }
+    else {
+        at86rf2xx_fb_stop(dev);
+    }
+
+    mutex_unlock(&dev->lock);
+    return pkt_len;
+}
+
+static int _request_on(ieee802154_dev_t *hal)
+{
+    DEBUG("at86rf2xx_rf_ops: request_on\n");
+    at86rf2xx_t *dev = hal->priv;
+    mutex_lock(&dev->lock);
+#if !AT86RF2XX_IS_PERIPH
+    gpio_irq_enable(dev->params.int_pin);
+#endif
+    /* enable interrupts */
+    at86rf2xx_reg_write(dev, AT86RF2XX_REG__IRQ_MASK,
+                        AT86RF2XX_IRQ_STATUS_MASK__TRX_END | AT86RF2XX_IRQ_STATUS_MASK__RX_START);
+
+
+#if AT86RF2XX_IS_PERIPH
+    /* Setting SLPTR bit in TRXPR to 0 returns the radio transceiver
+     * to the TRX_OFF state */
+    *AT86RF2XX_REG__TRXPR &= ~(AT86RF2XX_TRXPR_SLPTR);
+#else
+    gpio_clear(dev->params.sleep_pin);
+#endif
+
+    mutex_unlock(&dev->lock);
+    return 0;
+}
+
+static int _confirm_on(ieee802154_dev_t *hal)
+{
+    DEBUG("at86rf2xx_rf_ops: confirm_on\n");
+    at86rf2xx_t *dev = hal->priv;
+    mutex_lock(&dev->lock);
+    int status = at86rf2xx_reg_read(dev, AT86RF2XX_REG__TRX_STATUS)
+                     & AT86RF2XX_TRX_STATUS_MASK__TRX_STATUS;
```

Comment:
```suggestion
    int status = at86rf2xx_reg_read(dev, AT86RF2XX_REG__TRX_STATUS)
               & AT86RF2XX_TRX_STATUS_MASK__TRX_STATUS;
```
just a bit nicer to read